src/context/GlobalContext.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { createContext, useReducer, useEffect, ReactNode, Dispatch } from "react";
+
+import AppReducer from "./AppReducer";
+import {Item, ItemType, State, Action} from "../Objects"
+
+const initialItems: State = JSON.parse(localStorage.getItem("list") || "") || [];
+//  JSON.parse는 string을 인자로 받으나 localStorage의 리턴 타입은 string or null 이므로 ""을 추가한다
```

Comment:
저도 이렇게 했는데 초기에 로컬스토리지에 값이 없으면 에러가 나요 ㅠㅠ
지금 배포해주신 링크도 로컬스토리지 지우고 새로고침하면 에러나서 아예 아무것도 뜨지 않네요
아래 함수가 좀 이상하긴 한데, 저는 대충 저렇게 오류 해결했습니당 

    const getLocalValue = (listName = 'list') => {
    
      const value = localStorage.getItem(listName);
      let localData;
      if (typeof value === 'string') {
  
          localData = JSON.parse(value);
      }else if(value === null){
  
          localData = []
      }
  
      return localData
    };

Reply:
감사합니다 시원님, 말씀 주신대로 localStorage 삭제하니 에러가 나네요!
시원님 코드 참고해서 이렇게 수정했습니다. 공유해주셔서 감사해요~

```suggestion
const initialItems: State = JSON.parse(localStorage.getItem("list") || "[]");
//  JSON.parse는 string을 인자로 받으나 localStorage의 리턴 타입은 string or null 이므로 ""을 추가한다
```